lib/ui/window.dart

```diff
@@ -283,7 +283,24 @@ class Window {
   /// Message handlers must call the function given in the `callback` parameter.
   /// If the handler does not need to respond, the handler should pass `null` to
   /// the callback.
-  PlatformMessageCallback onPlatformMessage;
+  ///
+  /// The default handler queues incoming messages and relays them to
+  /// the first custom handler set, flushing the queue. The default handler is
+  /// restored, if [onPlatformMessage] is set to null.
+  PlatformMessageCallback get onPlatformMessage => _onPlatformMessage;
+  void set onPlatformMessage(PlatformMessageCallback callback) {
+    if (callback == null) {
+      _onPlatformMessage = _queueingPlatformMessageCallback;
+    } else {
+      if (_onPlatformMessage == _queueingPlatformMessageCallback) {
+        for (_PlatformMessage message in _messageQueue)
```

Comment:
Consider what happens if the `onPlatformMessage` setter is called during this iteration.  You probably want to make a local copy of the queue and clear it out before sending callbacks.

Also, calling the callback re-entrantly here makes a difficult-to-use API.  This setter is likely to be called some time during startup before all the receivers are fully registered.  Consider delaying deliver until a microtask.  (If you do that, you probably want to freeze the set of messages to be delivered when you schedule the microtask and consider the case where the setter is returned to null between the time the task is scheduled and when it executes.)